Applications/Spire/Include/Spire/Spire/SpinBoxModel.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef SPIRE_SPIN_BOX_MODEL_HPP
+#define SPIRE_SPIN_BOX_MODEL_HPP
+#include <QHash>
+#include "Nexus/Definitions/Money.hpp"
+#include "Nexus/Definitions/Quantity.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/RealSpinBox.hpp"
+
+namespace Spire {
+
+  //! Represents the data model for a spin box.
+  template <typename T>
+  class SpinBoxModel {
+    public:
+
+      //! Constructs a SpinBoxModel with no modifiers.
+      /*!
+        \param initial The initial value.
+        \param minimum The minimum acceptable value.
+        \param maximum The maximum acceptable value.
+      */
+      SpinBoxModel(T initial, T minimum, T maximum)
+        : m_initial(initial),
+          m_minimum(minimum),
+          m_maximum(maximum) {}
+
+      //! Returns the initial value.
+      T get_initial() const {
+        return m_initial;
+      }
+
+      //! Sets the initial value.
+      /*!
+        \param initial The initial value.
+      */
+      void set_initial(T initial) {
+        m_initial = initial;
+      }
+
+      //! Returns an increment value. Returns the NoModifer value if the
+      //! modifiers have no set increment.
+      /*
+        \param modifiers The modifiers to get the increment value of.
+      */
+      T get_increment(Qt::KeyboardModifiers modifiers) const {
+        if(m_modifiers.contains(modifiers)) {
+          return m_modifiers[modifiers];
+        }
+        return m_modifiers[Qt::NoModifier];
```

Comment:
When will execution ever go through line 49?

Reply:
If the user enters modifiers that aren't sent then it should return the NoModifier value.